code/datums/components/food/germ_sensitive.dm

```diff
@@ -116,9 +119,9 @@ GLOBAL_LIST_INIT(floor_diseases, list(
 	var/random_disease = pick_weight(GLOB.floor_diseases)
 	parent.AddComponent(/datum/component/infective, new random_disease, weak = TRUE)
 
-/datum/component/germ_sensitive/proc/wash()
+/datum/component/germ_sensitive/proc/delete_germs()
 	SIGNAL_HANDLER
-	if(infective)
+	if(!infective)
 		infective = FALSE
 		qdel(parent.GetComponent(/datum/component/infective))
 		return COMPONENT_CLEANED
```

Comment:
Is this `infective` check supposed to get inverted? It's a bit odd to set `infective = FALSE` and remove the infective component when `!infective` notes `infective` was already `FALSE`

Reply:
cool cool